### PR TITLE
bpf/sock: Use renamed field

### PR DESCRIPTION
**Pull request by jrajahalme**
Use the backend 'flags' field instead the old 'pad'. This fixes bpf compilation fails when L7 LB is used, like this:

```
level=warning msg="/var/lib/cilium/bpf/bpf_sock.c:387:14: error: no member named 'pad' in 'struct lb4_backend'" subsys=datapath-loader
level=warning msg="                        l7backend.pad = 0;" subsys=datapath-loader
level=warning msg="                        ~~~~~~~~~ ^" subsys=datapath-loader
level=warning msg="1 error generated." subsys=datapath-loader
```

Added `ENABLE_L7_LB` to `LB_OPTIONS` in `bpf/Makefile` to verify the fix and to catch errors like this in future.

Fixes: #18814
Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>

### CODE REVIEW

**Review by sayboras — APPROVED** [review]
LGTM :100:

**Review by pchaigno — APPROVED** [review]
Changes look good to me.

> Note that as of now CI will not trigger this bug nor verify this fix. Ingress job lands in the CI soon to catch errors like this.

Why? Don't we have compilation tests in `bpf/Makefile` (`LB_OPTIONS`) to capture exactly this?

**Review by borkmann — APPROVED** [review]

**Comment by jrajahalme** [comment]
> Changes look good to me.
> 
> > Note that as of now CI will not trigger this bug nor verify this fix. Ingress job lands in the CI soon to catch errors like this.
> 
> Why? Don't we have compilation tests in `bpf/Makefile` (`LB_OPTIONS`) to capture exactly this?

@pchaigno If so I would assume every PR would have failed on this once https://github.com/cilium/cilium/pull/18814 merged?
At least I did not know to add anything to `bpf/Makefile` to capture this when working on L7 LB.

**Comment by jrajahalme** [comment]
> > > Note that as of now CI will not trigger this bug nor verify this fix. Ingress job lands in the CI soon to catch errors like this.
> > 
> > 
> > Why? Don't we have compilation tests in `bpf/Makefile` (`LB_OPTIONS`) to capture exactly this?
> 
> @pchaigno If so I would assume every PR would have failed on this once #18814 merged? At least I did not know to add anything to `bpf/Makefile` to capture this when working on L7 LB.

@pchaigno Added `ENABLE_L7_LB` to `LB_OPTIONS` in `bpf/Makefile` to catch errors like this in future.

**Review by pchaigno — APPROVED** [review]
> @pchaigno Added ENABLE_L7_LB to LB_OPTIONS in bpf/Makefile to catch errors like this in future.

Nice!

File: bpf/bpf_sock.c [hunk 1]
```diff
@@ -384,7 +384,7 @@ static __always_inline int __sock4_xlate_fwd(struct bpf_sock_addr *ctx,
 			l7backend.address = bpf_htonl(0x7f000001);
 			l7backend.port = (__be16)svc->l7_lb_proxy_port;
 			l7backend.proto = 0;
-			l7backend.pad = 0;
+			l7backend.flags = 0;
```
Comment (aditighag) [inline bpf/bpf_sock.c:387]: I wonder if it's better to zero out the `l7backend` struct using memset right after it's declared earlier in the function?

Comment (pchaigno) [reply bpf/bpf_sock.c:387]: `struct lb4_backend l7backend = {0};` would also work if we want to use that approach.

Comment (aditighag) [reply bpf/bpf_sock.c:387]: AFAIK, `memset` sets padding bits to `0` as well. It doesn't really matter much in this case though.

**Review by christarazi — APPROVED** [review]

**Comment by pchaigno** [comment]
`BPF checks / build datapath` passed so we should be good to merge.

**Pull request merged**